### PR TITLE
chore(analytics): adding status code to connector Kafka events

### PR DESCRIPTION
**Pull request by harsh-sharma-juspay**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
adding status code to connector Kafka events

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
f
do payment confirm call and then check for the status_code in log in loki  where ```topic=hyperswitch-outgoing-connector-events```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by harsh-sharma-juspay** [comment]
Screenshot 2024-01-18 at 11 29 49 PM

**Comment by harsh-sharma-juspay** [comment]
Screenshot 2024-01-18 at 11 32 07 PM

File: crates/router/src/services/api.rs [hunk 1]
```diff
@@ -372,7 +372,13 @@ where
                     let response = call_connector_api(state, request).await;
                     let external_latency = current_time.elapsed().as_millis();
                     logger::debug!(connector_response=?response);
-
+                    let status_code = response
+                        .as_ref()
+                        .map(|i| {
+                            i.clone()
```
Comment (lsampras) [inline crates/router/src/services/api.rs:378]: Is clone needed here? Since status_code is a u8 and can be copied directly...

**Review by lsampras — DISMISSED** [review]

**Review by ivor-juspay — DISMISSED** [review]

File: crates/router/src/services/api.rs (hunk 1 above)
Comment (harsh-sharma-juspay) [reply crates/router/src/services/api.rs:378]: it becomes *i from ownership of response been transferred

**Review by lsampras — APPROVED** [review]

**Review by sagarnaikjuspay — APPROVED** [review]

**Review by ivor-juspay — APPROVED** [review]

**Pull request merged**